dtu_mlops_project/visualization/visualize.py

```diff
@@ -0,0 +1,57 @@
+import os
+
+import dotenv
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+import torch
+
+from dtu_mlops_project.models.predict_model import PreprocPlusModel
+
+dotenv.load_dotenv(override=True)
+
+
+def generate_prediction_figure(path_to_model_checkpoint, path_to_waveform):
+
+    ppm = PreprocPlusModel(path_to_model_checkpoint)
+    results = ppm.wav_to_yhat(path_to_waveform=path_to_waveform)
+
+    idx_to_class = np.load("data/processed/class_idx_export.npy").astype(str)
+    predicted_class = idx_to_class[results["pred"]]
+    waveform = results["waveform"]
+    spectrogram = results['x_preproc'][0][0]
+
+    # computing "confidence" of prediction.
+    y_hat_activation = torch.nn.LogSoftmax(dim=1)
+    y_hat = torch.exp(y_hat_activation(results["y_hat"])).detach().numpy()
+    y_hat = (y_hat*100).flatten() # to percentages
+    conf = str(np.round(y_hat[results["pred"]],3))
+
+
+
+    fig = plt.figure(figsize=(12,10), tight_layout=True)
+    gs = gridspec.GridSpec(5, 6)
+    ax = fig.add_subplot(gs[0:3, 0:3])
+    ax.set_title(f"waveform\ntrue_class: {results['y_true']}")
+    ax.plot(waveform[0][0])
+
+    ax = fig.add_subplot(gs[0:3,3:6])
+    ax.set_title(f"spectrogram\npredicted_class: {predicted_class}     (conf: {conf}%)")
+    ax.imshow(spectrogram, origin='lower')
```

Comment:
@miccio-dk Is origin='lower' correct here? by which I mean: do lower indexes after preprocessing represent lower frequencies? Also what is the physical unit of the colormap? is it decibels?

Reply:
yes, and yes :)